### PR TITLE
xdp-dump: add support for multiple buffer wakeup

### PR DESCRIPTION
**Pull request by chaudron**

### CODE REVIEW

**Comment by chaudron** [comment]
> There seem to be a lot of unrelated whitespace changes that got in here somehow. Could you try to get rid of those, please?

I cleaned up the file as there where tabs and space used. So made it all spaces... You want a separate commit for this?

**Comment by tohojo** [comment]
Yes, please. Makes it clearer what actually changed for the new check :)

**Comment by chaudron** [comment]
> Yes, please. Makes it clearer what actually changed for the new check :)

Done, added additional commit at the end.

File: xdp-dump/test-xdpdump.sh [hunk 1]
```diff
@@ -62,6 +64,10 @@ Options:
 END
 	  )
 
+    if grep -q "HAVE_LIBBPF_PERF_BUFFER__CONSUME:=n" "$CONFIG_MK"; then
```
Comment (tohojo) [inline xdp-dump/test-xdpdump.sh:67]: I would like to make the tests runnable from the installed RPM. In which case config.mk won't be available. So can we come up with a better way to detect this?

Some ideas:
- grep the --help output
- have the Makefile write a test config that we can install along with the test scripts

File: xdp-dump/xdpdump.c [hunk 2]
```diff
@@ -79,6 +81,12 @@ static struct prog_option xdpdump_options[] = {
 		      .short_opt = 'i',
 		      .metavar = "<ifname>",
 		      .help = "Name of interface to capture on"),
+#ifdef HAVE_LIBBPF_PERF_BUFFER__CONSUME
+	DEFINE_OPTION("perf-wakeup", OPT_U32, struct dumpopt, perf_wakeup,
+		      .short_opt = 4,
```
Comment (tohojo) [inline xdp-dump/xdpdump.c:86]: Are these non-printable short opts really needed? If so, I'd prefer to fix params.c so they aren't...

Comment (chaudron) [reply xdp-dump/xdpdump.c:86]: Yes, they are, else you would crash :( Your current code needs them.

Comment (tohojo) [reply xdp-dump/xdpdump.c:86]: Right, okay, let's just merge this as-is, and I'll look into fixing params.c later...

File: xdp-dump/test-xdpdump.sh (hunk 1 above)
Comment (tohojo) [reply xdp-dump/test-xdpdump.sh:67]: Didn't see any resolution of this?

Comment (chaudron) [reply xdp-dump/test-xdpdump.sh:67]: I missed this comment, let me change this, and ping you!

File: xdp-dump/xdpdump.c (hunk 2 above)
Comment (chaudron) [reply xdp-dump/xdpdump.c:86]: Don't forget you need to merge PcapNG first for the tests to work.

**Comment by chaudron** [comment]
Fixed test check to use --help option to determine if wakeup feature is present.

File: xdp-dump/test-xdpdump.sh [hunk 3]
```diff
@@ -3,10 +3,11 @@
 #
 # shellcheck disable=2039
 #
-ALL_TESTS="test_help test_interfaces test_capt_pcap test_capt_term test_exitentry test_snap test_none_xdp"
+ALL_TESTS="test_help test_interfaces test_capt_pcap test_capt_term test_exitentry test_snap test_perf_wakeup test_none_xdp"
 
 XDPDUMP=./xdpdump
 XDP_LOADER=../xdp-loader/xdp-loader
+CONFIG_MK=../config.mk
```
[the remark this answers is not in the document]
Comment (chaudron) [reply xdp-dump/test-xdpdump.sh:10]: oops, will fix in the next batch when I cleanup the short parameter.

**Comment by chaudron** [comment]
Last commit change fixed/removed "CONFIG_MK=../config.mk", and fixed short option.

**Pull request merged**